### DISCUSSION TITLE
Need for speed

### DISCUSSION THREAD

**Discussion by TimGeyssens** [TimGeyssens/Umbraco-v8-Community-Website-StarterKit#6] [General]
Best practices for front end performance:

- [Automatically serving WebP images](https://kieron.codes/blog/using-webp-images-in-umbraco/)
- ...

**Comment by NikRimington** [comment]
Utilise Slimsy for variable image sizes

**Reply by TimGeyssens** [reply]
yes!

**Reply by SteveVaneeckhout** [reply]
Every browser has native support for responsive images and lazy loading. You don't need Slimsy.

**Reply by JasonElkin** [reply]
The server-side component of Slimsy is great (though srcset support really should be in core) so +1 for that
The JS is opinionated and may not necessarily be the right choice.

**Comment by TimGeyssens** [comment]
Another one could be [Html Minifier](https://our.umbraco.com/packages/developer-tools/html-minifier/)

**Comment by TimGeyssens** [comment]
And also best practices for caching (web config stuff) https://github.com/h5bp/server-configs-iis/blob/master/dotnet%204.5/MVC5/Web.config

**Comment by sjorspa** [comment]
Keep in mind that WebP is not supported by Safari, but maybe it is possible to create a helper function to generate the right Image snippets.

**Reply by mastrup** [reply]
Would be nice with some sort of function that could generate a picture element with WebP and JPEG versions + media queries.

**Reply by TimGeyssens** [reply]
yes the blog post takes care of that, it checks the Context.Request.AcceptTypes.Contains("image/webp")

**Reply by Matthew-Wise** [reply]
I would also use the same handler to set upscale=false to reduce image blurring :)

blog post I assume @TimGeyssens  mentions - https://kieron.codes/blog/using-webp-images-in-umbraco/

**Reply by NikRimington** [reply]
Also, if we are using Slimsy, then that provides a setup approach for explictitly setting a fallback for unsupported browsers as well

**Comment by dawoe** [comment]
Maybe a Donut output caching.

And Query performance.... V8 is different from V7. For example what is the tipping point for doing Descendants... In V8 the impact is not as heavy as in V7... but it's still there.

**Comment by TimGeyssens** [comment]
Might be usefull: https://dev.to/ender_minyard/the-ultimate-guide-to-web-performance-ci4
